kubernetes/include/generic.h

```diff
@@ -14,6 +14,10 @@ typedef struct genericClient_t {
     char *resourcePlural;
 } genericClient_t;
 
+#if !defined(HAVE_STRNDUP)
```

Comment:
The files `generic.{h|c}` are used to define a generic(no need to know the detail of Kubernetes resources) API client.
They are not the suitable place to put this code.
Could you please find another solution ? maybe a new file ?